### ISSUE TITLE
Image changes not properly reflected if element is used inside another element

### ISSUE THREAD

**Issue by billmoser** [a1626/image-carousel#8]
As discussed [here](https://github.com/billmoser/image-carousel/commit/d0a601278b28df116215143432adc1c8479fb37e), there are a couple of  features I'd like to add (and I would set up the pull requests for these).  I have a problem though: the original demo does not seem to work.   Here's what I'm doing:
```
bower install
polymer build
polymer serve
```
Then, I visit 127.0.0.1:8081 in the browser.
The page does not render correctly - rather, this text is all that is displayed:
```
Basic image-carousel demo
image-carousel demo with autoplay feature
image-carousel demo with no dots
image-carousel demo with no navigation icons
```
I believe the problem is that the paths are wrong in demo/index.html, or am I missing something?  If this is working, then I can just extend to add to demo the new features.   BTW, the features are:

1. Using dom-repeat to populate carousel does not work from inside another polymer element.  The problem seems to be that `this.computeNumberOfImages()` is being called before the slot is populated.  Why this only happens inside another polymer element, and not directly in index.html, I don't know.  Adding the following to `connectedCallback` fixed the problem:
```
this.shadowRoot.querySelector('slot[attribute="images"]').addEventListener('slotchange', e => {
            this.computeNumberOfImages();
        });
```

2. There is currently no way to react to the selection of a particular image in the carousel.  For my use case, I need to do some back-end stuff based on a particular image. Dispatching the event following inside the `if(elem)` conditional of `_changeAttrAndProps` achieves this:
```
let event  = new CustomEvent("selection-changed" , {
       composed: true, // allows to pass through shadow dom to parent
       detail: { 
           id: this.id,
           index: e.model.index
   }});
this.dispatchEvent(event);
```
When the event is dispatched, any listeners get the id of the carousel element doing the dispatching (assuming that the id was set on the element, of course), as well as the index of the carousel item being displayed.

I'll be happy to do these as pull requests, but first the demo should be in working order.  Or maybe I need to be educated as to what I'm doing incorrectly in running it :)

**Comment by billmoser** [comment]
So, I _was_ missing something after all.  I was visiting the wrong url.  when I visit http://127.0.0.1:8081/components/image-carousel/demo, all is good!
If you want me to add the pull requests & update the demo, let me know...

**Comment by a1626** [comment]
Sure. That will be great

**Comment by a1626** [comment]
For your `selection-changed` is `image-carousel-changed` event not doing the trick?

**Comment by billmoser** [comment]
image-carousel-changed seems only to fire when images are inserted, not when selected by the user

**Comment by a1626** [comment]
Just to clarify you want to do some action when the image that is viewable changes and not when user selects/clicks on the image

**Comment by billmoser** [comment]
What I want is an event fired when the user selects an image, either with the dots or the arrows.  In my use case, I load the carousel once, using dom-repeat (as in the other demo cases).  What I see is that 'image-carousel-changed' fires three times, presumably as each image is loaded.  But that doesn't do me any good for my use case.  What I need is that the event fires when the user initiates the change, either through using the arrow keys, or the dots.  The 'selection-changed' event that I'm proposing does just that.

**Comment by a1626** [comment]
So, that does not include image change because of `auto-scroll`?

**Comment by billmoser** [comment]
Ah, good point. I've added a user-settable boolean, ``notifySelectionChanged``, whose default value is ``false``.  That way, the user must specify this explicitly in order for notifications to be generated.  A user could set ``nofitySelectionChanged=="true"`` if desired, even when autoscrolling, if that's the behavior they want.  This is not checked in yet -- do you want me to add it to the current pull request branch, or wait until you've merged that, at which time I'd do another pull request?

**Comment by a1626** [comment]
Another PR will be a better option.

**Comment by billmoser** [comment]
Ok, will do

**Comment by a1626** [comment]
I've published new version `2.1.0` with all the fixes.
